### ISSUE TITLE
First-level support for contributions

### ISSUE THREAD

**Issue by schmmd** [allenai/allennlp-demo#467]
Presently some of our models have contribution text, but it's not supported very cleanly.  I think we should add an optional contributedBy field to the model object so we can easily add contributions in a consistent manner.

**Comment by matt-gardner** [comment]
I think "contributed" is maybe also the wrong word. In what sense did Nitish "contribute" the NMN model?  I would probably have fields for "built by", "paper", and "code".

**Comment by schmmd** [comment]
He made it work with AllenNLP, created a Docker image, pushed it, and collaborated with us to get it live... my goal is to acknowledge who gets the model in the demo to encourage others to also do so.

**Comment by matt-gardner** [comment]
Yeah, I definitely agree that we should recognize contributions, I'm just saying that "contributed" is perhaps the wrong word here.  The model doesn't live in AllenNLP, it's just in the demo.  It's not "contributed" to AllenNLP, and if we use the word "contributed", it could be confusing, unless we're more clear.  You could imagine showing this:

```
Demo contributed by: Nitish Gupta (but really, Sam should go here too)
Paper introducing model: [link to paper]
Code for model: [link to Nitish's repo]
```

It gets a bit more confusing if one third party takes some other third party code and helps us make it into a demo.  It'd be more clear with fields like this:

```
Demo built by: Nitish Gupta and Sam Skjonsberg
Paper introducing model: [link to paper]
Model code written by: Nitish Gupta and Matt Gardner
Code for model: [link to Nitish's repo]
```
